demos/python_demos/human_pose_estimation_demo/human_pose_estimation.py

```diff
@@ -0,0 +1,285 @@
+                      help='Optional. List of monitors to show initially.')
+    return parser
+
+
+class Modes(Enum):
+    USER_SPECIFIED = 0
+    MIN_LATENCY = 1
+
+
+class ModeInfo:
+    def __init__(self):
+        self.last_start_time = perf_counter()
+        self.last_end_time = None
+        self.frames_count = 0
+        self.latency_sum = 0
+
+
+def get_plugin_configs(device, num_streams, num_threads):
+    config_user_specified = {}
+    config_min_latency = {}
+
+    devices_nstreams = {}
+    if num_streams:
+        devices_nstreams = {device: num_streams for device in ['CPU', 'GPU'] if device in device} \
+                           if num_streams.isdigit() \
+                           else dict(device.split(':', 1) for device in num_streams.split(','))
+
+    if 'CPU' in device:
+        if num_threads is not None:
+            config_user_specified['CPU_THREADS_NUM'] = str(num_threads)
+        if 'CPU' in devices_nstreams:
+            config_user_specified['CPU_THROUGHPUT_STREAMS'] = devices_nstreams['CPU'] \
+                                                              if int(devices_nstreams['CPU']) > 0 \
+                                                              else 'CPU_THROUGHPUT_AUTO'
+
+        config_min_latency['CPU_THROUGHPUT_STREAMS'] = '1'
+
+    if 'GPU' in device:
+        if 'GPU' in devices_nstreams:
+            config_user_specified['GPU_THROUGHPUT_STREAMS'] = devices_nstreams['GPU'] \
+                                                              if int(devices_nstreams['GPU']) > 0 \
+                                                              else 'GPU_THROUGHPUT_AUTO'
+
+        config_min_latency['GPU_THROUGHPUT_STREAMS'] = '1'
+
+    return config_user_specified, config_min_latency
+
+
+def main():
+    args = build_argparser().parse_args()
+
+    log.info('Initializing Inference Engine...')
+    ie = IECore()
+
+    config_user_specified, config_min_latency = get_plugin_configs(args.device, args.num_streams, args.num_threads)
+
+    log.info('Loading network...')
+    completed_request_results = {}
+    modes = cycle(Modes)
+    prev_mode = mode = next(modes)
+    log.info('Using {} mode'.format(mode.name))
+    mode_info = {mode: ModeInfo()}
+    exceptions = []
+
+    if args.type == 'ae':
+        HPE = HPEAssociativeEmbedding
+    else:
+        HPE = HPEOpenPose
+
+    hpes = {
+        Modes.USER_SPECIFIED:
+            HPE(ie, args.model, target_size=args.tsize, device=args.device, plugin_config=config_user_specified,
+                results=completed_request_results, max_num_requests=args.num_infer_requests,
+                caught_exceptions=exceptions),
+        Modes.MIN_LATENCY:
+            HPE(ie, args.model, target_size=args.tsize, device=args.device.split(':')[-1].split(',')[0],
+                plugin_config=config_min_latency, results=completed_request_results, max_num_requests=1,
+                caught_exceptions=exceptions)
```

Comment:
@Wovchena do we need two modes here?

Reply:
No, but @druzhkov-paul was pointed at your version when it had this. That means to me it's our responsibility now to rewrite it later (that same is about `-at, --architecture_type` and other things).